tensorflow/java/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -190,47 +224,85 @@ public static Tensor create(long[] shape, DoubleBuffer data) {
    * @param data a buffer containing the tensor data.
    * @throws IllegalArgumentException If the tensor shape is not compatible with the buffer
    */
-  public static Tensor create(long[] shape, LongBuffer data) {
-    Tensor t = allocateForBuffer(DataType.INT64, shape, data.remaining());
+  public static Tensor<Long> create(long[] shape, LongBuffer data) {
+    Tensor<Long> t = allocateForBuffer(DataType.INT64, shape, data.remaining());
     t.buffer().asLongBuffer().put(data);
     return t;
   }
 
   /**
-   * Create a Tensor with data from the given buffer.
+   * Create a Tensor of any type with data from the given buffer.
    *
    * <p>Creates a Tensor with the provided shape of any type where the tensor's data has been
    * encoded into {@code data} as per the specification of the TensorFlow <a
    * href="https://www.tensorflow.org/code/tensorflow/c/c_api.h">C API</a>.
    *
-   * @param dataType the tensor datatype.
+   * @param <T> the tensor element type
+   * @param type the tensor element type, represented as a class object.
    * @param shape the tensor shape.
    * @param data a buffer containing the tensor data.
    * @throws IllegalArgumentException If the tensor datatype or shape is not compatible with the
    *     buffer
    */
-  public static Tensor create(DataType dataType, long[] shape, ByteBuffer data) {
+  @SuppressWarnings("unchecked")
+  public static <T> Tensor<T> create(Class<T> type, long[] shape, ByteBuffer data) {
+    return (Tensor<T>) create(DataType.fromClass(type), shape, data);
+  }
+
+  /**
+   * Creates a Tensor of any type with data from the given buffer.
+   *
+   * <p>Creates a Tensor with the provided shape of any type where the tensor's data has been
+   * encoded into {@code data} as per the specification of the TensorFlow <a
+   * href="https://www.tensorflow.org/code/tensorflow/c/c_api.h">C API</a>.
+   *
+   * @param <T> The tensor element type
+   * @param type the tensor element type, specified as a DataType. This must agree with T.
+   * @param shape the tensor shape.
+   * @param data a buffer containing the tensor data.
+   * @throws IllegalArgumentException If the tensor datatype or shape is not compatible with the
+   *     buffer
+   */
+  public static Tensor<?> create(DataType dtype, long[] shape, ByteBuffer data) {
```

Comment:
Let's make this private (ensuring that all creation methods require a `Class` and not a `DataType`)?

Reply:
This is a preexisting method so I was leaving it around for backward compatibility. How about we deprecate it?

Reply:
I believe the only real user of this was `TensorFlowInferenceInterface`, which you've updated. So I'd be tempted to mark this private.